packages/tailwindcss/src/index.ts

```diff
@@ -501,6 +541,30 @@ async function parseCss(
     node.context = {}
   }
 
+  // Replace the `@variant` at-rules with the actual variant rules.
+  if (variantNodes.length > 0) {
+    for (let variantNode of variantNodes) {
+      // Starting with the `&` rule node
+      let node = styleRule('&', variantNode.nodes)
+
+      for (let variant of segment(variantNode.params, ':').reverse()) {
```

Comment:
To decide:

Right now this code means that we can use `@variant md:dark:hover {}`. If we _only_ want to allow a single `@variant`, then this could be simplified to just:
```suggestion
      for (let variant of [variantNode.params]) {
```

And get rid of the loop.